### PR TITLE
evm!: update go-ethereum dependency

### PR DESCRIPTION
**Pull request by dessaya**
EVM native contract addresses are now specified in the wasp repository.

BREAKING CHANGE:
- Magic contract address is now 0x1074000000000000000000000000000000000000
- ERC20 base tokens is now      0x1074010000000000000000000000000000000000

### CODE REVIEW

**Comment by muXxer** [comment]
Did you also update the go Ethereum version to a specific tag? Maybe we could tag our release instead of using the commit hash.

I did that last time as you can see in the go.mod changes :)

**Comment by dessaya** [comment]
> Did you also update the go Ethereum version to a specific tag?

I rebased my changes to the same v1.10.25 tag.

> Maybe we could tag our release instead of using the commit hash.

What would be the benefit of doing that?

File: packages/vm/core/evm/iscmagic/embed.go [hunk 1]
```diff
@@ -37,11 +43,15 @@ var (
...
 
-var ERC20BaseTokensAddress = iscAddressPlusOne()
+var ERC20BaseTokensAddress = makeMagicAddress(addressTypeERC20BaseTokens, nil)
 
-func iscAddressPlusOne() common.Address {
-	n := binary.BigEndian.Uint64(vm.ISCAddress.Bytes()[common.AddressLength-8:]) + 1
+func makeMagicAddress(kind byte, data []byte) common.Address {
 	var ret common.Address
-	binary.BigEndian.PutUint64(ret[common.AddressLength-8:], n)
+	if len(data) > len(ret)-3 {
```
Comment (muXxer) [inline packages/vm/core/evm/iscmagic/embed.go:50]:
```suggestion
	// 2 bytes "prefix" + 1 byte "kind"
	if len(data) > common.AddressLength-3 {
```

**Review by muXxer — APPROVED** [review]
LGTM

File: packages/vm/core/evm/iscmagic/embed.go (hunk 1 above)
Comment (dessaya) [reply packages/vm/core/evm/iscmagic/embed.go:50]: Thx, will apply this in my next PR

Comment (muXxer) [reply packages/vm/core/evm/iscmagic/embed.go:50]: yeah sorry I didn't see that you already enabled auto merge :)

**Pull request merged**